packages/material-ui/src/Tooltip/Tooltip.js

```diff
@@ -556,7 +562,11 @@ Tooltip.propTypes = {
    * Override or extend the styles applied to the component.
    * See [CSS API](#css) below for more details.
    */
-  classes: PropTypes.object.isRequired,
```

Comment:
Just a question for my own understanding: strictly speaking `Tooltip` requires `classes` to be present otherwise `classes.root` would blow up, but I guess
- we can always guarantee them being present due to `withStyles`
- from a user's point of view, they're not required and we are generating `PropTypes` from the user-facing `.d.ts`, correct?

That being said, if this component were to be written in TypeScript, then you'd probably have to define `TooltipProps` as having a non-optional `classes` (otherwise you'd need to check `classes ? ` each time), and then it's the application of `withStyles` HOC that would make the public facing API have an optional `classes`?

Reply:
TooltipProps are the props of the outer component. The propTypes are on the inner component. Yes, classes should be required here in propTypes but since these are passed from an implementation we control there's little value checking them. At least I can't imagine a scenario where this propType warning would help identify the issue.

Reply:
> TooltipProps are the props of the outer component.

Sorry, maybe my unfamiliarity with TS speaking - I thought that you would have to type the inner component's props, kinda what we're doing with `PropTypes` on the inner component.

Reply:
Yeah HOCs and outer vs inner props was always a bit problematic in TS. We don't consume any types though so everything you see is concerned with what you can put in i.e. outer props.